src/AutoRest.CSharp/Common/Output/Models/Types/EnumType.cs

```diff
@@ -58,15 +58,19 @@ public EnumType(InputEnumType input, string defaultNamespace, string defaultAcce
             ValueType = typeFactory.CreateType(input.EnumValueType);
             IsStringValueType = ValueType.Equals(typeof(string));
             IsIntValueType = ValueType.Equals(typeof(int)) || ValueType.Equals(typeof(long));
+            IsFloatValueType = ValueType.Equals(typeof(float));
```

Comment:
do we need to include `double` here? Could we ever have double value enum?
But even if we never have double value enum, it would not harm if we add it i assume

Reply:
According to the code in `TypeSpecInputEnumTypeConverter`, it can be only `int`, `float` or `string`. But yeah, I can add it.